README.md

```diff
@@ -1,11 +1,118 @@
 # infra-deployments
 
 ## Konflux Community Deployment via ArgoCD 
-```console
+```bash
 ./deploy-argocd.sh <ArgoCD server> <production/staging>
 ```
 
 ## Accessing the Konflux Console
-```console
+```bash
 kubectl port-forward -n konflux-ui svc/proxy 9443:9443 
-```
\ No newline at end of file
+```
+
+## Setting up a development environment using kind
+
+### Installing Software Dependencies
+Verify that the applications below are installed on the host machine:
+
+* [Kind and kubectl](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+  along with `podman` or `docker`
+* [ArgoCD cli](https://argo-cd.readthedocs.io/en/stable/cli_installation/)
+
+### Bootstrapping the development cluster
+Clone this repository:
+
+ ```bash
+git clone https://github.com/konflux-ci/infra-deployments
+cd infra-deployments
+```
+
+**Note:** It is recommended that you increase the `inotify` resource limits in order to
+avoid issues related to
+[too many open files](https://kind.sigs.k8s.io/docs/user/known-issues/#pod-errors-due-to-too-many-open-files). 
+To increase the limits temporarily, run the following commands:
+
+```bash
+sudo sysctl fs.inotify.max_user_watches=524288
+sudo sysctl fs.inotify.max_user_instances=512
+```
+
+From the root of this repository, run the setup scripts:
+
+1. Create a cluster
+
+```bash
+kind create cluster --name dev --config kind-config.yaml
+```
+
+**Note:** When using Podman, it is recommended that you increase the PID limit on the
+container running the cluster, as the default might not be enough when the cluster
+becomes busy:
+
+```bash
+podman update --pids-limit 4096 dev-control-plane
+```
+
+**Note:** If pods still fail to start due to missing resources, you may need to reserve
+additional resources to the Kind cluster. Edit [kind-config.yaml](./kind-config.yaml)
+and modify the `system-reserved` line under `kubeletExtraArgs`:
+
+```yaml
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+        system-reserved: memory=12Gi
+```
+
+2. Create argocd namespace in the cluster:
+```bash
+kubectl create namespace argocd
+```
+
+3. Deploy ArgoCD:
+```bash
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+
+4. Once ArgoCD is up and running, obtain its server's password:
+```bash
+kubectl -n argocd get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" | base64 -d && echo
+```
+This password will be used to login to ArgoCD UI as well as for deploying konflux.
+
+5. To deploy konflux, we need to have access to ArgoCD server, we will port forward ArgoCD server:
+```bash
+kubectl port-forward -n argocd service/argocd-server 8443:443
+```
+
+6. Comment the line the `login_argocd "$1"` in the main method of the `deploy-argocd.sh` script as follows:
+```bash
+main() {
+    verify_permissions || exit $?
+#   login_argocd "$1"
```

Comment:
Maybe we can change the line (in git) so it won't fail on failed login. Something like this:
```
login_argocd "$1" | true
```
Then we can ask the user to login beforehand like you suggest here, and when running the script, this line will fail but the failure will be ignored and the script will keep running.

We can verify in some other way that the user is logged in after that line (although I'm not sure it's necessary for now).

Reply:
I thought of adding another parameter `skip` or `skip-login`
And in this case we can run the script as we do now, but if e add this parameter, we will skip the login.
So we will be able to run it as follows:
`./deploy-argocd.sh localhost:8443 staging` to allow login
And 
`./deploy-argocd.sh localhost:8443 staging skip` to skip login

WDYT?

Reply:
Not sure it's needed. The script will fail soon enough if the login was unsuccessful. I suggest we go with just the `| true| addition and see if we run into any issues later on.